l10n_it_website_portal_fatturapa/controllers/main.py

```diff
@@ -0,0 +1,90 @@
+# Copyright 2019 Simone Rubino
+# Copyright 2019 Lorenzo Battistini
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo.exceptions import ValidationError
+from odoo.http import request
+from odoo.tools.translate import _
+
+from odoo.addons.l10n_it_website_portal_fiscalcode.controllers.main import (
+    WebsitePortalFiscalCode,
+)
+
+FATTURAPA_PORTAL_FIELDS = [
+    "codice_destinatario",
+    "firstname",
+    "lastname",
+    "pec_destinatario",
+    "country_id",
+    "fiscalcode",
+    "zipcode",
+    "vat",
+    "electronic_invoice_subjected",
+    "street",
+    "city",
+    "is_pa",
+    "ipa_code",
+    "eori_code",
+    "electronic_invoice_obliged_subject",
+]
+WebsitePortalFiscalCode.OPTIONAL_BILLING_FIELDS.extend(FATTURAPA_PORTAL_FIELDS)
+
+
+class WebsitePortalFatturapa(WebsitePortalFiscalCode):
+    def details_form_validate(self, data):
+        # when checkbox electronic_invoice_subjected is not checked,
+        # it is not posted
+        data["electronic_invoice_subjected"] = data.get(
+            "electronic_invoice_subjected", False
+        )
+        if data["electronic_invoice_subjected"]:
```

Comment:
Cosa ne pensi di trasformare questo blocco in

        if data["electronic_invoice_subjected"]:
            data["electronic_invoice_obliged_subject"] = True
            country_italy = request.env.ref('base.it', raise_if_not_found=False)
            country_id = data.get('country_id', False)
            if country_id and country_italy and int(country_id) != country_italy.id:
                data["codice_destinatario"] = "XXXXXXX"
            elif data.get('codice_destinatario') == "XXXXXXX":
                data["codice_destinatario"] = "0000000"

? per via di questo https://github.com/OCA/l10n-italy/pull/2520#issuecomment-1733203982

Reply:
fatto

Reply:
@Borruso Grazie, un ultimo appunto è che si potrebbe importare il codice destinatario da qui
https://github.com/OCA/l10n-italy/blob/14.0/l10n_it_fatturapa/models/partner.py#L7 , ma non so se ne vale la pena, decidete voi 👍